external-crates/move/crates/move-vm-runtime/src/execution/interpreter/eval.rs

```diff
@@ -43,13 +44,15 @@ enum StepStatus {
     Done,
 }
 
-struct RunContext<'vm_cache, 'native, 'native_lifetimes> {
+struct RunContext<'vm_cache, 'native, 'native_lifetimes, 'tracer, 'trace_builder> {
     vtables: &'vm_cache VMDispatchTables,
     vm_config: Arc<VMConfig>,
     extensions: &'native mut NativeContextExtensions<'native_lifetimes>,
+    // TODO: consider making this `Option<&mut VMTracer<'_>>` and passing it like that everywhere?
+    tracer: &'tracer mut Option<VMTracer<'trace_builder>>,
```

Comment:
That might be better, as the tracer would live between commands in the PTB layer. Although how much better I'm not sure -- I think if it were an `Option<&mut ..>` it might be a bit easier to deal with but not _that_ much easier.